pkg/kubectl/cmd/util/openapi/dryrun.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"errors"
+
+	openapi_v2 "github.com/googleapis/gnostic/OpenAPIv2"
+	yaml "gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func checkExtension(extensions []*openapi_v2.NamedAny, gvk schema.GroupVersionKind) bool {
+	for _, extension := range extensions {
+		if extension.GetValue().GetYaml() == "" ||
```

Comment:
I'm not familiar with this method of getting data out of the openapi doc... it's yaml-encoded?

cc @mbohlool for review

Reply:
Since vendor extensions can be any type of yaml data in the openapi spec, this framework is just keeping it as bytes that can be decoded into whatever needed.

Reply:
Antonie used gnostic openapi structs more than me, if this works, I am fine with it. If I want to be on the safe side, I will only check the name and if the value is empty, log error or fail or something because that is not usual.